### PR TITLE
Backend: More tests

### PR DESCRIPTION
**Pull request by OxygenCobalt**
Add more and better tests to the CRUD routes that aren't likely to change very much (At least after #82). Will undraft as soon as #82 is merged, since it changes the alert code a bit.

- Use an in-memory mock database for testing. This is pretty quick and also allows us to make sure our queries are doing the right thing
- Add some equality and repr methods for the database models so I can confirm that no data is being mangled in transit.

### CODE REVIEW

**Comment by shanecranor** [comment]
LGTM I'll let Tyler review before we merge

**Review by tbwrigh — APPROVED** [review]
LGTM

**Pull request merged**
